tools/config/__init__.py

```diff
@@ -692,12 +692,13 @@ def _assign_new_offset(rom_start, start, new_offset, region_name):
         newstart = rom_start + integer(new_offset, 0)
         if newstart < start:
             raise ConfigException(
-                "Can not place % region inside previous region" % region_name)
+                "Can not place %r region inside previous region" % region_name)
         return newstart
 
     def _generate_bootloader_build(self, rom_start, rom_size):
         start = rom_start
         rom_end = rom_start + rom_size
+        max_app_addr = -1
```

Comment:
Please don't use `-1` as a marker. Pythonic optional types are generally `None` or an instance of the type. Could you replace this `-1` with `None`?